test/e2e_env/multizone/gateway/cross-mesh.go

```diff
@@ -99,29 +115,55 @@ func CrossMeshGatewayOnMultizone() {
 	})
 
 	Context("when mTLS is enabled", func() {
-		It("should proxy HTTP requests from a different mesh in the same zone", func() {
-			gatewayAddr := net.JoinHostPort(crossMeshHostname, strconv.Itoa(crossMeshGatewayPort))
-			k8s_gateway.SuccessfullyProxyRequestToGateway(
-				env.KubeZone1, gatewayMesh,
-				gatewayAddr,
-				gatewayClientNamespaceOtherMesh,
-			)
+		gatewayAddr := net.JoinHostPort(crossMeshHostname, strconv.Itoa(crossMeshGatewayPort))
+		Context("Intrazone", func() {
+			zone := &env.KubeZone1
+			It("proxies HTTP requests from a different mesh", func() {
+				k8s_gateway.SuccessfullyProxyRequestToGateway(
+					*zone, gatewayMesh,
+					gatewayAddr,
+					gatewayClientNamespaceOtherMesh,
+				)
+			})
+			It("proxies HTTP requests from the same mesh", func() {
+				k8s_gateway.SuccessfullyProxyRequestToGateway(
+					*zone, gatewayMesh,
+					gatewayAddr,
+					gatewayClientNamespaceSameMesh,
+				)
+			})
 		})
-		It("should proxy HTTP requests from the same mesh in the same zone", func() {
-			gatewayAddr := net.JoinHostPort(crossMeshHostname, strconv.Itoa(crossMeshGatewayPort))
-			k8s_gateway.SuccessfullyProxyRequestToGateway(
-				env.KubeZone1, gatewayMesh,
-				gatewayAddr,
-				gatewayClientNamespaceSameMesh,
+		Context("Crosszone", func() {
+			zone := &env.KubeZone2
+			filter := fmt.Sprintf(
+				"cluster.%s_%s.upstream_rq_total",
+				gatewayMesh,
+				crossMeshGatewayName,
 			)
-		})
-		It("doesn't proxy HTTP requests from another zone yet", func() {
-			gatewayAddr := net.JoinHostPort(crossMeshHostname, strconv.Itoa(crossMeshGatewayPort))
-			Consistently(k8s_gateway.FailToProxyRequestToGateway(
-				env.KubeZone2,
-				gatewayAddr,
-				gatewayClientNamespaceOtherMesh,
-			), "10s", "1s").Should(Succeed())
+			It("proxies HTTP requests from a different mesh", func() {
+				egress := (*zone).GetZoneEgressEnvoyTunnel()
+				Expect(egress.ResetCounters()).To(Succeed())
+
+				k8s_gateway.SuccessfullyProxyRequestToGateway(
+					*zone, gatewayMesh,
+					gatewayAddr,
+					gatewayClientNamespaceOtherMesh,
+				)
+
+				Expect(egress.GetStats(filter)).To(stats.BeGreaterThanZero())
+			})
+			It("proxies HTTP requests from the same mesh", func() {
+				egress := (*zone).GetZoneEgressEnvoyTunnel()
+				Expect(egress.ResetCounters()).To(Succeed())
+
+				k8s_gateway.SuccessfullyProxyRequestToGateway(
+					*zone, gatewayMesh,
+					gatewayAddr,
+					gatewayClientNamespaceOtherMesh,
+				)
+
+				Expect(egress.GetStats(filter)).To(stats.BeGreaterThanZero())
+			})
```

Comment:
aren't those test cases exactly the same?

Reply:
Fixed